src/screens/SignUp/SignUp.js

```diff
@@ -1,14 +1,75 @@
-import React from "react";
-import { Link } from "react-router-dom";
-import "./SignUp.css";
+import React, { useState } from "react";
+import { Link, useNavigate } from "react-router-dom";
+import { userSignUp } from "../../services";
 
 const SignUp = () => {
+	const navigate = useNavigate();
+	const [username, setUsername] = useState("");
+	const [email, setEmail] = useState("");
+	const [password, setPassword] = useState("");
+	const [passwordCheck, setPasswordCheck] = useState("");
+	const [termsAndConditions, setTermsAndConditions] = useState(false);
+	const [error, setError] = useState("");
+
+	const handleFormFill = (data, setFunction) => {
+		if (error) setError("");
+		setFunction(data);
+	};
+
+	const handleSignUp = async (event) => {
+		event.preventDefault();
+		try {
```

Comment:
for checking validation you can create a separate file that can have all the validation maintained of your project

Reply:
I'll do that